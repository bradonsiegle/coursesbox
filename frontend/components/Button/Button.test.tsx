import React from 'react';
import { render, screen } from '@/test-utils';
import userEvent from '@testing-library/user-event';

import {
	Button,
	PrimaryButton,
	SecondaryButton,
	DangerButton,
	WarningButton,
} from './Button';

describe('Button test cases', () => {
	it('should render', () => {
		const onClick = jest.fn();
		const { asFragment } = render(<Button onClick={onClick}>Test</Button>);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should render primary button', () => {
		const onClick = jest.fn();
		const { asFragment } = render(
			<PrimaryButton onClick={onClick}>Test</PrimaryButton>
		);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should render secondary button', () => {
		const onClick = jest.fn();
		const { asFragment } = render(
			<SecondaryButton onClick={onClick}>Test</SecondaryButton>
		);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should render danger button', () => {
		const onClick = jest.fn();
		const { asFragment } = render(
			<DangerButton onClick={onClick}>Test</DangerButton>
		);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should render warning button', () => {
		const onClick = jest.fn();
		const { asFragment } = render(
			<WarningButton onClick={onClick}>Test</WarningButton>
		);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should call onClick callback', async () => {
		const onClick = jest.fn();
		render(<Button onClick={onClick}>Test</Button>);
		const button = screen.getByRole('button');
		await userEvent.click(button);
		expect(onClick).toHaveBeenCalled();
	});
});
