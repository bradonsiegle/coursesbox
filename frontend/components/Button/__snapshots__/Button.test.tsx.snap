// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases should render 1`] = `
<DocumentFragment>
  <button
    class="css-4dagrr"
    color="primary"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render danger button 1`] = `
<DocumentFragment>
  <button
    class="css-u7ws8f"
    color="danger"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render primary button 1`] = `
<DocumentFragment>
  <button
    class="css-4dagrr"
    color="primary"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render secondary button 1`] = `
<DocumentFragment>
  <button
    class="css-110k448"
    color="secondary"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render warning button 1`] = `
<DocumentFragment>
  <button
    class="css-1rbl8ym"
    color="warning"
  >
    Test
  </button>
</DocumentFragment>
`;
